diff --git a/link/.spacemacs b/link/.spacemacs
index 47e1440..f626803 100644
--- a/link/.spacemacs
+++ b/link/.spacemacs
@@ -33,6 +33,12 @@ This function should only modify configuration layer settings."
    ;; List of configuration layers to load.
    dotspacemacs-configuration-layers
    '(
+     csv
+     javascript
+     python
+     html
+     sql
+     theming
      ;; ----------------------------------------------------------------
      ;; Example of useful layers you may want to use right away.
      ;; Uncomment some layer names and press `SPC f e R' (Vim style) or
@@ -44,13 +50,15 @@ This function should only modify configuration layer settings."
      ;; git
      helm
      ;; lsp
-     ;; markdown
+     markdown
      multiple-cursors
-     ;; org
+     (org :variables
+          search-invisible t
+          org-display-inline-images nil
+          org-startup-with-inline-images nil
+          org-src-preserve-indentation t ;; Fix issue with code blocks indentation increasing every time I press <return>
+          org-enable-reveal-js-support t)
      ;; (shell :variables
      ;;        shell-default-height 30
      ;;        shell-default-position 'bottom)
-     ;; spell-checking
+     (spell-checking :variables
+                     ispell-program-name "aspell"
+                     spell-checking-enable-by-default t)
      ;; syntax-checking
      ;; version-control
      treemacs)
@@ -501,7 +509,7 @@ It should only modify the values of Spacemacs settings."
    ;; `trailing' to delete only the whitespace at end of lines, `changed' to
    ;; delete only whitespace for changed lines or `nil' to disable cleanup.
    ;; (default nil)
-   dotspacemacs-whitespace-cleanup nil
+   dotspacemacs-whitespace-cleanup trailing
 
    ;; If non-nil activate `clean-aindent-mode' which tries to correct
    ;; virtual indentation of simple modes. This can interfere with mode specific
@@ -551,6 +559,22 @@ This function is called immediately after `dotspacemacs/init', before layer
 configuration.
 It is mostly for variables that should be set before packages are loaded.
 If you are unsure, try setting them in `dotspacemacs/user-config' first."
+
+  ;; Set a more discreet font-size variation for org headers
+  ;; https://emacs.stackexchange.com/questions/22584/disable-enlarged-org-mode-header-appearance
+  ;;(setq theming-modifications
+  ;;      '((spacemacs-light
+  ;;         (org-level-1 :height 1.15)
+  ;;         (org-level-2 :height 1.10)
+  ;;         (org-level-3 :height 1.05)
+  ;;         (org-level-4 :height 1.0)
+  ;;         (org-level-5 :height 1.0))
+  ;;        (spacemacs-dark
+  ;;         (org-level-1 :height 1.15)
+  ;;         (org-level-2 :height 1.10)
+  ;;         (org-level-3 :height 1.05)
+  ;;         (org-level-4 :height 1.0)
+  ;;         (org-level-5 :height 1.0))))
 )
 
 
@@ -568,6 +592,81 @@ This function is called at the very end of Spacemacs startup, after layer
 configuration.
 Put your configuration code here, except for variables that should be set
 before packages are loaded."
+  ;; ---------------------------------------------------------------------------
+  ;; Essential Vim behaviour
+  ;; ---------------------------------------------------------------------------
+
+  ;; Set escape keybinding to "jj" (except when in visual mode)
+  (setq-default evil-escape-key-sequence "jj"
+                evil-escape-delay 0.2
+                evil-escape-excluded-states '(visual))
+
+  ;; Prevent kill and yank commands from accessing the system clipboard
+  (setq select-enable-clipboard nil)
+
+  ;; Display the visited file's path in the frame title
+  ;; https://github.com/syl20bnr/spacemacs/issues/8613
+  (setq frame-title-format
+        '((:eval (if (buffer-file-name)
+                     (abbreviate-file-name (buffer-file-name))
+                   "%b"))))
+
+  ;; Revert to the normal behaviour for S in vim-surround
+  (evil-define-key 'visual evil-surround-mode-map "s" 'evil-substitute)
+  (evil-define-key 'visual evil-surround-mode-map "S" 'evil-surround-region)
+
+  ;; Have Ctrl+A and Ctrl+X increment and decrement numbers, as in Vim (does not work for dates, unfortunately)
+  ;; https://emacs.stackexchange.com/questions/41919/have-ctrl-a-increase-a-number-just-like-in-vim
+  (define-key evil-normal-state-map (kbd "C-a") 'evil-numbers/inc-at-pt)
+  (define-key evil-visual-state-map (kbd "C-a") 'evil-numbers/inc-at-pt)
+  (define-key evil-normal-state-map (kbd "C-x") 'evil-numbers/dec-at-pt)
+  (define-key evil-visual-state-map (kbd "C-x") 'evil-numbers/dec-at-pt)
+
+
+  ;; ---------------------------------------------------------------------------
+  ;; Custom key mappings
+  ;; ---------------------------------------------------------------------------
+
+  ;; Y yanks the current inner word into the "y register
+  (define-key evil-normal-state-map (kbd "Y") (kbd "\"yyiw"))
+  ;; S replaces the current inner word with the "y register
+  (define-key evil-normal-state-map (kbd "S") (kbd "diw\"yP"))
+
+  ;; Mappings behind SPC o (see also the "Saved macros" section)
+  (spacemacs/declare-prefix "o" "Lucian's shortcuts")
+
+  (spacemacs/declare-prefix "oy" "Yank to primary clipboard")
+  (spacemacs/set-leader-keys "oy" (kbd "\"+y"))
+  (spacemacs/declare-prefix "od" "Delete to primary clipboard")
+  (spacemacs/set-leader-keys "od" (kbd "\"+d"))
+  (spacemacs/declare-prefix "op" "Paste from primary clipboard after cursor")
+  (spacemacs/set-leader-keys "op" (kbd "\"+p"))
+  (spacemacs/declare-prefix "oP" "Paste from primary clipboard before cursor")
+  (spacemacs/set-leader-keys "oP" (kbd "\"+P"))
+
+  ;; Saved macros (behind SPC o o)
+  (spacemacs/declare-prefix "oo" "Macros")
+  (spacemacs/declare-prefix "oou" "Remove org link, keep URL")
+  (spacemacs/set-leader-keys "oou" (kbd "EB2xkjf]d2f]")) ;; the kj sequence is there to give a chance to org-mode to "refresh" the line and display the URL
+  (spacemacs/declare-prefix "oot" "Remove org link, keep text")
+  (spacemacs/set-leader-keys "oot" (kbd "EB2xkjdf[f]2x"))
+  (spacemacs/declare-prefix "ooa" "Convert org link to JIRA syntax")
+  (spacemacs/set-leader-keys "ooa" (kbd "EB2xkjdf]f]pxF]r|"))
+  (spacemacs/declare-prefix "ood" "Convert org link to Markdown syntax")
+  (spacemacs/set-leader-keys "ood" (kbd "EB2xkjdf]f]pxr)F]a(jj"))
+
+
+  ;; ---------------------------------------------------------------------------
+  ;; Org mode configuration
+  ;; ---------------------------------------------------------------------------
+
+  ;; Map the cut-subtree command, which apparently is not mapped by default
+  (spacemacs/set-leader-keys-for-major-mode 'org-mode "Sd" 'org-cut-subtree)
+
+  ;; FIXME: Workaround to have easy templates working again
+  ;; https://github.com/syl20bnr/spacemacs/issues/11798
+  (when (version<= "9.2" (org-version))
+    (require 'org-tempo))
 )
 
 
