--- .spacemacs.original	2024-07-25 19:15:34
+++ .spacemacs.mine	2024-07-25 19:09:00
@@ -2,6 +2,15 @@
 ;; This file is loaded by Spacemacs at startup.
 ;; It must be stored in your home directory.
 
+(defun lucian/run-dotfile-applescript (script-filename)
+  (let ((script (concat "run script \""
+                        (substitute-in-file-name "$HOME/dotfiles/apple-scripts/")
+                        script-filename
+                        "\"")))
+    (message "Executing AppleScript: %s" script)
+    (do-applescript script)))
+
+
 (defun dotspacemacs/layers ()
   "Layer configuration:
 This function should only modify configuration layer settings."
@@ -33,6 +42,12 @@
    ;; List of configuration layers to load.
    dotspacemacs-configuration-layers
    '(
+     csv
+     javascript
+     python
+     html
+     sql
+     theming
      ;; ----------------------------------------------------------------
      ;; Example of useful layers you may want to use right away.
      ;; Uncomment some layer names and press `SPC f e R' (Vim style) or
@@ -44,13 +59,23 @@
      ;; git
      helm
      ;; lsp
-     ;; markdown
+     markdown
      multiple-cursors
-     ;; org
+     (org :variables
+          search-invisible t
+          org-display-inline-images nil
+          org-startup-with-inline-images nil
+          org-src-preserve-indentation t ;; Fix issue with code blocks indentation increasing every time I press <return>
+          org-enable-reveal-js-support t
+          org-use-sub-superscripts '{} ;; Disable sub/super scrips using _word and ^word, but allow _{word} and ^{word}
+          org-export-with-sub-superscripts '{} ;; Like org-use-sub-superscripts, but for exporting
+          )
      ;; (shell :variables
      ;;        shell-default-height 30
      ;;        shell-default-position 'bottom)
-     ;; spell-checking
+     (spell-checking :variables
+                     ispell-program-name "aspell"
+                     spell-checking-enable-by-default t)
      ;; syntax-checking
      ;; version-control
      treemacs)
@@ -237,8 +262,8 @@
    ;; List of themes, the first of the list is loaded when spacemacs starts.
    ;; Press `SPC T n' to cycle to the next theme in the list (works great
    ;; with 2 themes variants, one dark and one light)
-   dotspacemacs-themes '(spacemacs-dark
-                         spacemacs-light)
+   dotspacemacs-themes '(spacemacs-light
+                         spacemacs-dark)
 
    ;; Set the theme for the Spaceline. Supported themes are `spacemacs',
    ;; `all-the-icons', `custom', `doom', `vim-powerline' and `vanilla'. The
@@ -259,7 +284,7 @@
    ;; a non-negative integer (pixel size), or a floating-point (point size).
    ;; Point size is recommended, because it's device independent. (default 10.0)
    dotspacemacs-default-font '("Source Code Pro"
-                               :size 10.0
+                               :size 13.0
                                :weight normal
                                :width normal)
 
@@ -515,7 +540,7 @@
    ;; `trailing' to delete only the whitespace at end of lines, `changed' to
    ;; delete only whitespace for changed lines or `nil' to disable cleanup.
    ;; (default nil)
-   dotspacemacs-whitespace-cleanup nil
+   dotspacemacs-whitespace-cleanup trailing
 
    ;; If non-nil activate `clean-aindent-mode' which tries to correct
    ;; virtual indentation of simple modes. This can interfere with mode specific
@@ -565,6 +590,9 @@
 configuration.
 It is mostly for variables that should be set before packages are loaded.
 If you are unsure, try setting them in `dotspacemacs/user-config' first."
+
+  ;; Enable Evil mode in the minibuffer too
+  (setq evil-want-minibuffer t)
 )
 
 
@@ -582,6 +610,104 @@
 configuration.
 Put your configuration code here, except for variables that should be set
 before packages are loaded."
+  ;; ---------------------------------------------------------------------------
+  ;; Global key bindings
+  ;; ---------------------------------------------------------------------------
+
+  ;; Mappings for my MacOS Service shortcuts
+  (global-set-key (kbd "C-s-<f10>")
+                  (lambda ()
+                    (interactive)
+                    (lucian/run-dotfile-applescript "volume-down.scpt")))
+  (global-set-key (kbd "C-M-s-<f10>")
+                  (lambda ()
+                    (interactive)
+                    (lucian/run-dotfile-applescript "volume-mute-unmute.scpt")))
+  (global-set-key (kbd "C-s-<f11>")
+                  (lambda ()
+                    (interactive)
+                    (lucian/run-dotfile-applescript "volume-up.scpt")))
+  ;; For the following to work, we need to let Emacs control "System Events" in the MacOS Security & Privacy Settings
+  ;; To achieve that, run the following command in the Emacs shell:
+  ;; osascript ~/dotfiles/apple-scripts/start-screen-saver.scpt
+  (global-set-key (kbd "C-s-<f12>")
+                  (lambda ()
+                    (interactive)
+                    (lucian/run-dotfile-applescript "start-screen-saver.scpt")))
+
+
+  ;; ---------------------------------------------------------------------------
+  ;; Essential Vim behaviour
+  ;; ---------------------------------------------------------------------------
+
+  ;; Set escape keybinding to "jj" (except when in visual mode)
+  (setq-default evil-escape-key-sequence "jj"
+                evil-escape-delay 0.2
+                evil-escape-excluded-states '(visual))
+
+  ;; Prevent kill and yank commands from accessing the system clipboard
+  (setq select-enable-clipboard nil)
+
+  ;; Display the visited file's path in the frame title
+  ;; https://github.com/syl20bnr/spacemacs/issues/8613
+  (setq frame-title-format
+        '((:eval (if (buffer-file-name)
+                     (abbreviate-file-name (buffer-file-name))
+                   "%b"))))
+
+  ;; Revert to the normal behaviour for S in vim-surround
+  (evil-define-key 'visual evil-surround-mode-map "s" 'evil-substitute)
+  (evil-define-key 'visual evil-surround-mode-map "S" 'evil-surround-region)
+
+  ;; Have Ctrl+A and Ctrl+X increment and decrement numbers, as in Vim (does not work for dates, unfortunately)
+  ;; https://emacs.stackexchange.com/questions/41919/have-ctrl-a-increase-a-number-just-like-in-vim
+  (define-key evil-normal-state-map (kbd "C-a") 'evil-numbers/inc-at-pt)
+  (define-key evil-visual-state-map (kbd "C-a") 'evil-numbers/inc-at-pt)
+  (define-key evil-normal-state-map (kbd "C-S-a") 'evil-numbers/dec-at-pt)
+  (define-key evil-visual-state-map (kbd "C-S-a") 'evil-numbers/dec-at-pt)
+
+
+  ;; ---------------------------------------------------------------------------
+  ;; Custom key mappings
+  ;; ---------------------------------------------------------------------------
+
+  ;; Y yanks the current inner word into the "y register
+  (define-key evil-normal-state-map (kbd "Y") (kbd "\"yyiw"))
+  ;; S replaces the current inner word with the "y register
+  (define-key evil-normal-state-map (kbd "S") (kbd "diw\"yP"))
+
+  ;; Mappings behind SPC o (see also the "Saved macros" section)
+  (spacemacs/declare-prefix "o" "Lucian's shortcuts")
+
+  (spacemacs/declare-prefix "oy" "Yank to primary clipboard")
+  (spacemacs/set-leader-keys "oy" (kbd "\"+y"))
+  (spacemacs/declare-prefix "od" "Delete to primary clipboard")
+  (spacemacs/set-leader-keys "od" (kbd "\"+d"))
+  (spacemacs/declare-prefix "op" "Paste from primary clipboard after cursor")
+  (spacemacs/set-leader-keys "op" (kbd "\"+p"))
+  (spacemacs/declare-prefix "oP" "Paste from primary clipboard before cursor")
+  (spacemacs/set-leader-keys "oP" (kbd "\"+P"))
+
+  ;; Saved macros (behind SPC o o)
+  (spacemacs/declare-prefix "oo" "Macros")
+  (spacemacs/declare-prefix "oou" "Remove org link, keep URL")
+  (spacemacs/set-leader-keys "oou" (kbd "EB2xkjf]d2f]")) ;; the kj sequence is there to give a chance to org-mode to "refresh" the line and display the URL
+  (spacemacs/declare-prefix "oot" "Remove org link, keep text")
+  (spacemacs/set-leader-keys "oot" (kbd "EB2xkjdf[f]2x"))
+  (spacemacs/declare-prefix "ooa" "Convert org link to JIRA syntax")
+  (spacemacs/set-leader-keys "ooa" (kbd "EB2xkjdf]f]pxF]r|"))
+  (spacemacs/declare-prefix "ood" "Convert org link to Markdown syntax")
+  (spacemacs/set-leader-keys "ood" (kbd "EB2xkjdf]f]pxr)F]a(jj"))
+
+
+  ;; ---------------------------------------------------------------------------
+  ;; Org mode configuration
+  ;; ---------------------------------------------------------------------------
+
+  ;; FIXME: Workaround to have easy templates working again
+  ;; https://github.com/syl20bnr/spacemacs/issues/11798
+  (when (version<= "9.2" (org-version))
+    (require 'org-tempo))
 )
 
 
